docs/tutorial/introduction.md

```diff
@@ -0,0 +1,74 @@
+# Introduction to decomp
+
+In this project, we are decompiling The Legend of Zelda: Ocarina of Time. This means that we take the assembly language that is on the cartridge,
+
+```
+glabel func_80A13098
+/* 00028 80A13098 8482001C */  lh      $v0, 0x001C($a0)           ## 0000001C
+/* 0002C 80A1309C 24010004 */  addiu   $at, $zero, 0x0004         ## $at = 00000004
+/* 00030 80A130A0 14410003 */  bne     $v0, $at, .L80A130B0       
+/* 00034 80A130A4 244EFFFE */  addiu   $t6, $v0, 0xFFFE           ## $t6 = FFFFFFFE
+/* 00038 80A130A8 10000002 */  beq     $zero, $zero, .L80A130B4   
+/* 0003C 80A130AC A480001C */  sh      $zero, 0x001C($a0)         ## 0000001C
+.L80A130B0:
+/* 00040 80A130B0 A48E001C */  sh      $t6, 0x001C($a0)           ## 0000001C
+.L80A130B4:
+/* 00044 80A130B4 8C8F0330 */  lw      $t7, 0x0330($a0)           ## 00000330
+/* 00048 80A130B8 24020001 */  addiu   $v0, $zero, 0x0001         ## $v0 = 00000001
+/* 0004C 80A130BC 24180011 */  addiu   $t8, $zero, 0x0011         ## $t8 = 00000011
+/* 00050 80A130C0 A1E20004 */  sb      $v0, 0x0004($t7)           ## 00000004
+/* 00054 80A130C4 A08201B8 */  sb      $v0, 0x01B8($a0)           ## 000001B8
+/* 00058 80A130C8 A08201B9 */  sb      $v0, 0x01B9($a0)           ## 000001B9
+/* 0005C 80A130CC 03E00008 */  jr      $ra                        
+/* 00060 80A130D0 A0980117 */  sb      $t8, 0x0117($a0)           ## 00000117
+```
+
+(the commented numbers on the left are the original machine code, the middle the translation into MIPS assembly, the right useful information about the numbers in the code)
+and turn it into compilable C code:
+
+```C
+void func_80A13098(EnFirefly* this) {
+    if (this->actor.params == 4) {
+        this->actor.params = 0;
+    } else {
+        this->actor.params -= 2;
+    }
+    this->collider.list->body.toucher.effect = 1;
+    this->auraType = 1;
+    this->onFire = 1;
+    this->actor.naviEnemyId = 0x11;
+}
+```
+
+which is intended to be as close to the original code as possible. We are doing *matching* decomp: in the right context, and with the right compiler settings, the above C compiles into *precisely* the assembly code above, not just equivalent code.
+
+N.B. We are using only publicly available code. In particular, we are not looking at any of the recent Nintendo source code leaks. (These apparently contain very little Ocarina of Time material anyway.)
+
+Progress of the project can be found at [https://zelda64.dev]. The long-term goal of this project is to obtain a complete compilable version of the code for every publicly released version of Ocarina of Time (there are also sister projects for Majora's Mask and other Zelda games). *We are not working on a PC Port, and this project will not be making one*, although the resulting code will be very useful if someone does intend to make such a port.
```

Comment:
Why is the link to zelda64.dev in brackets?

Reply:
This is markdown for a bare link without diferent text: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links